pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -866,10 +866,8 @@ impl RemoteTimelineClient {
                 "Found {} files not bound to index_file.json, proceeding with their deletion",
                 remaining.len()
             );
-            for file in remaining {
-                warn!("Removing {}", file.object_name().unwrap_or_default());
-                self.storage_impl.delete(&file).await?;
-            }
+            warn!("About to remove {} files", remaining.len());
+            self.storage_impl.delete_objects(&remaining).await?;
```

Comment:
When reading this patch, I was starting to think about partial failure handling.
E.g., what happens in the following cases

1. `delete_objects([A, B, C])` but [`B`] does not exist on the remote
2. `delete_objects([A, B, C])` but we don't have permission to delete [`B`]
3. ???

I guess even before this patch, we were bailing out on first error, and I assume there's a higher-level retry mechanism.

So, feel free to dismiss this comment.

Reply:
when objects don't exist on the remote , S3 will return success 
and yes we bailed on the first error